packages/loopring_v3/contracts/impl/libexchange/ExchangeBlocks.sol

```diff
@@ -59,227 +60,115 @@ library ExchangeBlocks
-                uint8(blockType),
-                S.onchainDataAvailability,
-                blockSize,
-                blockVersion,
-                publicInputs,
-                proofs
-            ),
-            "INVALID_PROOF"
+            S.loopring.canExchangeCommitBlocks(S.id, S.onchainDataAvailability),
+            "INSUFFICIENT_EXCHANGE_STAKE"
         );
 
-        // Mark the blocks as verified
-        for (uint i = 0; i < blockIndices.length; i++) {
-            uint blockIdx = blockIndices[i];
-            ExchangeData.Block storage specifiedBlock = S.blocks[blockIdx];
-            // Check this again to make sure no block is verified twice in a single call to verifyBlocks
-            require(
-                specifiedBlock.state == ExchangeData.BlockState.COMMITTED,
-                "BLOCK_VERIFIED_ALREADY"
+        // Commit the blocks
+        bytes32[] memory publicDataHashes = new bytes32[](blocks.length);
+        for (uint i = 0; i < blocks.length; i++) {
+            // Hash all the public data to a single value which is used as the input for the circuit
+            publicDataHashes[i] = blocks[i].data.fastSHA256();
+            // Commit the block
+            commitBlock(
+                S,
+                blocks[i],
+                feeRecipient,
+                publicDataHashes[i]
             );
-            specifiedBlock.state = ExchangeData.BlockState.VERIFIED;
-            emit BlockVerified(blockIdx);
-        }
-
-        // Update the number of blocks that are finalized
-        // The number of blocks after the specified block index is limited
-        // by MAX_GAP_BETWEEN_FINALIZED_AND_VERIFIED_BLOCKS
-        // so we don't have to worry about running out of gas in this loop
-        uint idx = S.numBlocksFinalized;
-        while (idx < S.blocks.length &&
-            S.blocks[idx].state == ExchangeData.BlockState.VERIFIED) {
-            emit BlockFinalized(idx);
-            idx++;
         }
-        S.numBlocksFinalized = idx;
-    }
-
-    function revertBlock(
-        ExchangeData.State storage S,
-        uint blockIdx
-        )
-        external
-    {
-        // Exchange cannot be in withdrawal mode
-        require(!S.isInWithdrawalMode(), "INVALID_MODE");
-
-        require(blockIdx < S.blocks.length, "INVALID_BLOCK_IDX");
-        ExchangeData.Block storage specifiedBlock = S.blocks[blockIdx];
-        require(specifiedBlock.state == ExchangeData.BlockState.COMMITTED, "INVALID_BLOCK_STATE");
-
-        // The specified block needs to a non-finialized one.
-        require(blockIdx >= S.numBlocksFinalized, "FINALIZED_BLOCK_REVERT_PROHIBITED");
 
-        // Fine the exchange
-        uint fine = S.loopring.revertFineLRC();
-        S.loopring.burnExchangeStake(S.id, fine);
-
-        // Remove all blocks after and including blockIdx
-        S.blocks.length = blockIdx;
-
-        emit Revert(blockIdx);
+        // Verify the blocks
+        verifyBlocks(
```

Comment:
The order doesn't matter as any error will throw which fails the complete transaction so no blocks are submitted.

The reason to do the verification after the committing is because it makes testing easier as this allows creating invalid data for the block commitment easier without first already running into invalid block verifications because the OCDA was changed.